src/model/model.nit

```diff
@@ -1586,6 +1662,7 @@ class MNullType
 	redef fun full_name do return "null"
 	redef fun c_name do return "null"
 	redef fun as_nullable do return self
+	redef fun as_notnull do abort # sorry...
```

Comment:
Maybe you should add a comment that warns the user?

Or at least print something.